src/main/scala/com/fulcrumgenomics/bam/IdentifyPrimers.scala

```diff
@@ -0,0 +1,1027 @@
+
+import scala.collection.immutable
+import scala.reflect.runtime.{universe => ru}
+
+@clp(group=ClpGroups.SamOrBam, description=
+  """
+    |Identifies primers that generate the reads in the input BAM file.
+    |
+    |Note: when comparing a primer sequence to a read sequence, it "matches" if the two sequences are within at most
+    |`--max-mismatches` to the assigned primers with at most `--max-mismatch-delta` difference between between number
+    |of mismatches in the best and second best primer matches.
+    |
+    |## Primers
+    |
+    |The input primers file should contain a header, with one row per primer, as follows:
+    |
+    |  * pair_id - the unique identifier for the primer pair
+    |  * primer_id - the unique identifier for the primer in a primer pair
+    |  * sequence - the DNA sequence, including degenerate bases, in the sequencing order (i.e. if forward is False, the
+    |               sequence should be the reverse complement of the forward genomic strand).
+    |  * chr - the refNameosome/refName
+    |  * start - the start position (1-based)
+    |  * end - the end position (1-based, inclusive)
+    |  * forward - True if the forward primer, false otherwise
+    |
+    |The primer file must contain headers, e.g:
+    |
+    |```
+    |pair_id	primer_id	sequence	chr	start	end	forward
+    |1	1_1	GATTACA	chr1	1010873	1010894	True
+    |1	1_2	ACATTAG	chr1	1011118	1011137	True
+    |```
+    |
+    |Note: a canonical primer pair is the pair of primers with the same `pair_id`.  Other possible pairs are called
+    |"non-canonical".
+    |
+    |## Paired-End Matching (5' only)
+    |
+    |If the input are paired-end reads, then the reads are matched as follows for each read pair:
+    |
+    |1. Assign to a canonical primer pair if the 5' end of the mates agree with the mapping
+    |location of their respective primers.
+    |2. Assign to a non-canonical primer pair if the 5' end of the mates agree with the mapping location of their
+    |respective primers, and the primer and read sequences "match" respectively for each primer.
+    |3. Assign each mate of a read pair to a primer based on fully-gapped alignment.
+    |
+    |## Paired-End Matching (5' and 3')
+    |
+    |The `--three-prime` option can be used to also search the 3' end of every read for a primer sequence as follows:
+    |
+    |1. If a primer pair is identified using the 5' end, that pair is searched on the 3' end.
+    |2. Otherwise, assign a primer based on fully-gapped alignment.
+    |
+    |The primer matched on the 5' end of read one must match the primer matched on the 3' end of read two, and the
+    |primer matched on the 3' end of read one must match the primer matched on the 5' end of read two.  This will only
+    |be enforced if both ends of a read match a primer.
+    |
+    |## Fragment Matching (5' only)
+    |
+    |If the input are fragment reads, then only the 5' end of a read is used to match against the input list of primers:
```

Comment:
This disagrees with the next section.  Maybe you meant "when only" instead of "then only"?